### PR TITLE
feat(tests): Add integration tests for rhc connect with features config

### PR DESCRIPTION
**Pull request by DuckBoss**
This patch aims to add integration tests for the implementation of drop-in configuration file support for connectivity features. The tests cover cases ranging from default usage,
CLI/config features overriding, and edge cases with partial config files and feature dependency resolution.

Resolves: CCT-1973

AI Usage Disclosure: This commit contains test cases that were created with the assistance of AI tools.

### CODE REVIEW

File: integration-tests/test_connect.py [hunk 1]
```diff
@@ -876,3 +876,436 @@ def test_connect_with_nonexistent_feature(external_candlepin, rhc, test_config,
...
+            {"content": True, "analytics": True, "remote_management": True},
+            False,
+        ),
+        # Config enables analytics/remote-management, CLI disables content - dependency conflict (exit 64)
+        (
+            'features = { "content" = true, "analytics" = true, "remote-management" = true }',
+            [],
+            ["content"],
+            None,  # No expected states - command should fail
+            True,
```
Comment (Archana-PandeyM) [inline integration-tests/test_connect.py]:
```suggestion
            64,
```

File: integration-tests/test_connect.py [hunk 2]
```diff
@@ -876,3 +876,436 @@ def test_connect_with_nonexistent_feature(external_candlepin, rhc, test_config,
...
+        ),
+        # Config enables remote-management, CLI disables analytics - dependency conflict (exit 64)
+        (
+            'features = { "content" = true, "analytics" = true, "remote-management" = true }',
+            [],
+            ["analytics"],
+            None,  # No expected states - command should fail
+            True,
```
Comment (Archana-PandeyM) [inline integration-tests/test_connect.py]:
```suggestion
            64,
```

File: integration-tests/test_connect.py [hunk 3]
```diff
@@ -876,3 +876,436 @@ def test_connect_with_nonexistent_feature(external_candlepin, rhc, test_config,
...
+        test_config, auth="activation-key", output_format="json"
+    )
+
+    # Add CLI flags to override config
+    for feature in cli_enable:
+        command_args.extend(["--enable-feature", feature])
+    for feature in cli_disable:
+        command_args.extend(["--disable-feature", feature])
+
+    command = ["connect"] + command_args
```
Comment (Archana-PandeyM) [inline integration-tests/test_connect.py]: result = rhc.run("connect", *command_args, check=(expected_return_code == 0))

    assert result.returncode == expected_return_code, (
        f"Expected return code {expected_return_code}, but got {result.returncode}"
    )

    if expected_return_code != 0:
        # Dependency conflict: CLI disables a feature that config-enabled features depend on
        assert not rhc.is_registered, (
            "System should not be registered when CLI flag conflicts with config file"
        )
        return

    assert rhc.is_registered

    # Parse JSON output and verify feature states
    json_output = json.loads(result.stdout)
    features = json_output["features"]

    for feature_key, expected_enabled in expected_states.items():
        actual_enabled = features[feature_key]["enabled"]
        assert actual_enabled is expected_enabled, (
            f"Feature '{feature_key}' should have enabled={expected_enabled} "
            f"(CLI override), but got enabled={actual_enabled}"
        )

    # Verify yggdrasil service state
    if expected_states.get("remote_management", False):
        assert yggdrasil_service_is_active()
    else:
        assert not yggdrasil_service_is_active()

Comment (Archana-PandeyM) [reply integration-tests/test_connect.py]: I am suggesting this above code block for rest of test code. I could not find a way to suggest multiline code .

**Comment by Archana-PandeyM** [comment]
Thanks for the PR Jason. Overall it looks good.

Non-blocking Suggestion: split config-focused connect tests into a dedicated module to keep test_connect.py maintainable.
for example- Create integration-tests/test_connect_config_file.py 

I would like to hear your thoughts too.

**Comment by Archana-PandeyM** [comment]
I think it would be better to add some negative tests to cover the malformed toml file, so that errors should not be silently ignored.
e.g.
1- Invalid TOML
2- Unknown feature key
